components/raftstore-v2/src/operation/ready/snapshot.rs

```diff
@@ -596,12 +592,15 @@ impl<EK: KvEngine, ER: RaftEngine> Storage<EK, ER> {
             "{}",
             SlogFormat(self.logger())
         );
-        let region_state = self.region_state_mut();
+        let mut region_state = self.region_state().clone();
         region_state.set_state(PeerState::Normal);
         region_state.set_region(region);
         region_state.set_removed_records(removed_records);
         region_state.set_merged_records(merged_records);
         region_state.set_tablet_index(last_index);
+        // We need set_region_state here to update the peer.
+        self.set_region_state(region_state);
```

Comment:
I don't get it. So if L599 already change the state in place, why need to set it again in L263?

Reply:
`set_region_state` does not just set the region_state, it also update the peer.